app/main.py

```diff
@@ -1,2 +1,209 @@
+
+        while (
+            self.hash_table[index] is not None
+            and (
+                self.hash_table[index].key != key
+                or self.hash_table[index].key_hash != key_hash
+            )
+        ):
+            index = (index + 1) % self.capacity
+
+        return index
+
+    @property
+    def current_max_size(self) -> int:
+        return int(self.capacity * self.RESIZE_THRESHOLD)
+
+    def _resize(self) -> None:
+        old_hash_table = self.hash_table
+        self.capacity *= self.CAPACITY_MULTIPLIER
+        self.hash_table = [None] * self.capacity
+        self.size = 0
+
+        for node in filter(None, old_hash_table):
+            self.__setitem__(node.key, node.value)
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.size + 1 > self.current_max_size:
+            self._resize()
+
+        index = self._calculate_index(key)
+        key_hash = hash(key)
+
+        if self.hash_table[index] is None:
+            self.size += 1
+
+        self.hash_table[index] = Dictionary.Node(key, value, key_hash)
+
+    def __getitem__(self, key: Hashable) -> Any:
+        index = self._calculate_index(key)
+
+        if self.hash_table[index] is None:
+            raise KeyError(f"Cannot find value for key: {key}")
+
+        return self.hash_table[index].value
+
+    def __delitem__(self, key: Hashable) -> None:
+        index = self._calculate_index(key)
+
+        if self.hash_table[index] is None:
+            raise KeyError(f"Cannot delete value for key: {key}")
+
+        self.hash_table[index] = None
+        self.size -= 1
+
+        if self.size < self.capacity // self.CAPACITY_MULTIPLIER:
+            self._resize()
+
+    def __iter__(self) -> Iterator[tuple[Hashable, Any]]:
+        for node in self.hash_table:
+            if node is not None:
+                yield node.key, node.value
+
+    def __len__(self) -> int:
+        return self.size
+
+    def __str__(self) -> str:
+        items = [
+            f"{node.key}: {node.value}"
+            for node in self.hash_table
+            if node is not None
+        ]
+        return "{" + ", ".join(items) + "}"
+
+    def __eq__(self, other: Dictionary) -> bool:
+        """
+        Compare key, value pairs between two Dictionaries.
+        """
+        if not isinstance(other, Dictionary):
+            return False
+
+        for key, value in self:
+            if other.__getitem__(key) != value:
+                return False
+        return True
+
+    def get(self, key: Hashable, default: Optional[Any] = None) -> Any:
+        """
+        Get the value for the given key.
+        If the key does not exist, return the default value.
+
+        Default value is optional.
+        If it was not provided, method will raise KeyError.
+        """
+        try:
+            return self[key]
+        except KeyError:
+            return default
+
+    def pop(self, key: Hashable, default: Optional[Any] = None) -> Any:
```

Comment:
for correct implementation you need to pass object() as default value and check for that. None is a valid parameter for this method, it shouldnt trigger a KeyError